core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -105,7 +105,7 @@ private[spark] class DiskBlockManager(
         val newDir = new File(localDirs(dirId), "%02x".format(subDirId))
         if (!newDir.exists()) {
           val path = newDir.toPath
-          Files.createDirectory(path)
+          Files.createDirectories(path)
```

Comment:
Just for doubly sure, so if the local dir (e.g., `spark.local.dir`) is removed, this will create the local dir back. Is this behaviour consistent with other places that uses local dir? How does YARN remove this directory?